### ISSUE TITLE
Unable to get qz.printers.clearQueue() function working in Windows

### ISSUE THREAD

**Issue by joonshakya** [qzind/tray#1305]
When I create a print job and the printer is offline, I want to clear the job so that the print wont get executed when the printer is back online. To solve that, I keep note of the jobId and after 15 seconds after its in printing stage, if the print is not complete, I try to clear the job.

I am using code

```ts
qz.printers.clearQueue({
  printerName: "Bill",
  jobId: 19
});
```
In mac, this code works and job is cancelled, but in windows, it throws the error

Job# 19 is not part of the 'Bill' print queue
And if I just try to pass the name for qz-tray to clear all queue using this code, it doesn't perform any action

```ts
qz.printers.clearQueue({
  printerName: "Bill",
});
```
Here the printerName and the jobId are passed with correct values.

Minimal Code:
```ts
qz.websocket.connect().then(async () => {
    qz.printers.startListening(null);
    qz.printers.setPrinterCallbacks(async (e) => {
        if (e.eventType !== "JOB") {
            return;
        }
        if (e.statusText === "PRINTING") {
            console.log("Event Detals: ", JSON.stringify(e, null, 2))
            await qz.printers.clearQueue({
                printerName: e.printerName,
                jobId: e.jobId
            });
            console.log('Job successfully cancled');
        }
    });
    const listOfPrinters = await qz.printers.find();
    if (!listOfPrinters[0]) {
        throw new Error("No Printers Found");
    }
    console.log("Printers: ", JSON.stringify(listOfPrinters, null, 2))
    const config = qz.configs.create(
        {
            name: listOfPrinters[0],
        },
        {
            jobName: "hello",
        },
    );
    await qz.print(config, ["Hello world"]);
});
```

Output:
![main](https://github.com/user-attachments/assets/db77bb2f-5823-4797-a4e0-0dfc64ecc113)

**Comment by tresf** [comment]
Hi, thanks for the detailed explanation.  We should be able to reproduce with the steps provided.  While you await our reply, I have one question... does the behavior change if you toggle the option in the Windows print driver to "Print directly to the printer"?

<img width="429" alt="image" src="https://github.com/user-attachments/assets/5084cbc7-f3e7-4cc3-8f00-840f086ddd58">

**Comment by tresf** [comment]
@joonshakya so I was able to reproduce this in part... These are my steps:

1. Create a printer queue that goes nowhere (e.g. TCP/IP to an invalid IP address)
2. Submit two print jobs to this printer
3. The second job will remain queued, it CAN be cancelled (e.g. "Job 6")
4. The first job will go into failure state and CANNOT be cancelled (e.g. "Job 5").

With regards to this working on CUPS, cups has a different code path as can be observed here.  Notice that we only attempt to cancel jobs that are NOT in a status of RETAINED and are NOT in a status of PRINTED.  There's a chance that failed jobs are triggering one of these flags.  I will need to run a debugger to find out if this is the case.  We'll look into this.

https://github.com/qzind/tray/blob/b3a0e60bc6e85d7154c56de83f8ab88b07f95aee/src/qz/utils/PrintingUtilities.java#L275-L290

**Comment by tresf** [comment]
In older versions of Windows, the `RETAINED` flag would indicate that it could not be deleted. This is documented here:  https://learn.microsoft.com/en-us/windows/win32/printdocs/job-info-1

Quoting:


> `JOB_STATUS_RETAINED`	Windows Vista and later: Job has been retained in the print queue and cannot be deleted. This can be caused by the following:
> 1. The job was manually retained by a call to SetJob and the spooler is waiting for the job to be released.
> 2. The job has not finished printing and must finish printing before it can be automatically deleted.
See [SetJob](https://learn.microsoft.com/en-us/windows/win32/printdocs/setjob) for more information about print job commands.

... however testing this on Windows 11, this statement appears to be untrue.  Since we already log exceptions to QZ Tray (we do NOT report them to the API), it seems like a better strategy to simply send the `JOB_CONTROL_DELETE` blindly.  This will be patched in QZ Tray 2.2.5 and later versions.   To obtain a one-off copy in the interim, please ask and we'll upload one for testing purposes.
